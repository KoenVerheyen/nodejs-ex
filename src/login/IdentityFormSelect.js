// @flow
import React from "react";
import {Select} from "antd";

const Option = Select.Option

export type Element = {
    id: string,
    name? : string,
    firstName? : string,
    lastName? : string,
}

type Props = {
    name: string,
    value?: string | number,
    onSelectChange: (value: string) => void,
    elements: Array<Element>,
    getName: (elem: Element) => string,
    style: Object,
}

class IdentityFormSelect extends React.Component<Props> {


  render() {
    return (
      <Select name={this.props.name}
              value={this.props.value ? this.props.value : ''}
              onChange={this.props.onSelectChange}
              style={this.props.style}>
        {this.props.elements.map((elem) => {
          return (
            <Option key={elem.id} value={elem.id}>{this.props.getName(elem)}</Option>
          )
        })}
      </Select>
    )
  }
}

export default IdentityFormSelect