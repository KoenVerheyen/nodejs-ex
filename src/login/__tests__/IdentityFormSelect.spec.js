import React from 'react'
import {shallow} from 'enzyme'
import IdentityFormSelect from "../IdentityFormSelect";

describe('IdentityFormSelect', () => {

    test('basic with no value set', () => {
        let onChange = jest.fn()
        let wrapper = shallow(<IdentityFormSelect
            name='person'
            onSelectChange={onChange}
            elements={[{id: 1, firstName: 'John', lastName: 'Doe'}, {id:2, firstName: 'Jack', lastName: 'Doe'}]}
            getName={(elem) => elem.firstName + ' ' + elem.lastName}
        />)
        expect(wrapper.length).toEqual(1)
        expect(wrapper).toMatchSnapshot()

    })

    test('basic with value set', () => {
        let onChange = jest.fn()
        let wrapper = shallow(<IdentityFormSelect
            name='person'
            value={1}
            onSelectChange={onChange}
            elements={[{id: 1, firstName: 'John', lastName: 'Doe'}, {id:2, firstName: 'Jack', lastName: 'Doe'}]}
            getName={(elem) => elem.firstName + ' ' + elem.lastName}
        />)
        expect(wrapper.length).toEqual(1)
        expect(wrapper).toMatchSnapshot()

    })
})