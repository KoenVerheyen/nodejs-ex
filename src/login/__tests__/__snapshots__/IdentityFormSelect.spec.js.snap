// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IdentityFormSelect basic with no value set 1`] = `
<Select
  choiceTransitionName="zoom"
  name="person"
  onChange={[Function]}
  prefixCls="ant-select"
  showSearch={false}
  transitionName="slide-up"
  value=""
>
  <Option
    key="1"
    value={1}
  >
    John Doe
  </Option>
  <Option
    key="2"
    value={2}
  >
    Jack Doe
  </Option>
</Select>
`;

exports[`IdentityFormSelect basic with value set 1`] = `
<Select
  choiceTransitionName="zoom"
  name="person"
  onChange={[Function]}
  prefixCls="ant-select"
  showSearch={false}
  transitionName="slide-up"
  value={1}
>
  <Option
    key="1"
    value={1}
  >
    John Doe
  </Option>
  <Option
    key="2"
    value={2}
  >
    Jack Doe
  </Option>
</Select>
`;
